### PR TITLE
Clean-up JAAS authentication

### PR DESCRIPTION
**Pull request by kwart**
This PR is related to cleaning-up JAAS authentication in Hazelcast Enterprise:
* JAAS `Principal` cleanup
  * Separate authenticated name, its roles, and its origin
  * don't store Credentials within the `ClusterPrincipal` object
* improve work with `Callbacks`
  * let the `ClusterCallbackHandler` work with standard callback types (`NameCallback`, `PasswordCallback`)
  * introduce additional callbacks to give LoginModules access to some Hazelcast internals (Config object, serialization service, remote endpoint address);  don't abuse LoginModule options to provide these objects

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/security/SimpleTokenCredentials.java [hunk 1]
```diff
@@ -16,59 +16,82 @@
...
     }
 
     @Override
-    public final void setEndpoint(String endpoint) {
-        this.endpoint = endpoint;
+    public final void readPortable(PortableReader reader) throws IOException {
+        token = reader.readByteArray("token");
+    }
+
+    public int getFactoryId() {
```
Comment (galibey) [inline hazelcast/src/main/java/com/hazelcast/security/SimpleTokenCredentials.java]: Looks like @Override is missing here

File: hazelcast/src/main/java/com/hazelcast/security/UsernamePasswordCredentials.java [hunk 2]
```diff
@@ -16,83 +16,60 @@
...
-    protected void readPortableInternal(PortableReader reader) throws IOException {
-        password = reader.readByteArray("pwd");
+    /**
+     * @param name the name to set
+     */
+    public void setName(String name) {
+        this.name = name;
     }
 
     public int getFactoryId() {
```
Comment (galibey) [inline hazelcast/src/main/java/com/hazelcast/security/UsernamePasswordCredentials.java]: Here is also "Override" is missing here

File: hazelcast/src/main/java/com/hazelcast/security/SimpleTokenCredentials.java (hunk 1 above)
Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/security/SimpleTokenCredentials.java]: :+1: Added.

File: hazelcast/src/main/java/com/hazelcast/security/UsernamePasswordCredentials.java (hunk 2 above)
Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/security/UsernamePasswordCredentials.java]: :+1: Added.

**Comment by kwart** [comment]
Thanks for the review, @galibey . The missing `@Overrides` were added.

**Comment by kwart** [comment]
Thanks for the review, @tkountis.

**Pull request merged**